alchemy/src/docker/container.ts

```diff
@@ -0,0 +1,267 @@
+ * const appContainer = await Container("app", {
+ *   image: customImage, // Using an Alchemy RemoteImage resource
+ *   environment: {
+ *     NODE_ENV: "production",
+ *     API_KEY: "secret-key"
+ *   },
+ *   volumes: [
+ *     { hostPath: "./data", containerPath: "/app/data" }
+ *   ],
+ *   ports: [
+ *     { external: 3000, internal: 3000 }
+ *   ],
+ *   restart: "always",
+ *   start: true
+ * });
+ */
+export const Container = Resource(
+  "docker::Container",
+  async function (
+    this: Context<Container>,
+    id: string,
+    props: ContainerProps,
+  ): Promise<Container> {
+    // Initialize Docker API client
+    const api = new DockerApi();
+
+    // Get image reference
+    const imageRef =
+      typeof props.image === "string" ? props.image : props.image.imageRef;
+
+    // Use provided name or generate one based on resource ID
+    const containerName =
+      props.name || `alchemy-${id.replace(/[^a-zA-Z0-9_.-]/g, "-")}`;
+
+    // Handle delete phase
+    if (this.phase === "delete") {
+      if (this.output?.id) {
+        // Stop container if running
+        await api.stopContainer(this.output.id);
+
+        // Remove container
+        await api.removeContainer(this.output.id, true);
+      }
+
+      // Return destroyed state
+      return this.destroy();
+    } else {
+      let containerState: NonNullable<Container["state"]> = "created";
+
+      if (this.phase === "update") {
+        // Check if container already exists (for update)
+        const containerExists = await api.containerExists(containerName);
+
+        if (containerExists) {
+          // Remove existing container for update
+          await api.removeContainer(containerName, true);
+        }
+      }
+
+      // Prepare port mappings
+      const portMappings: Record<string, string> = {};
+      if (props.ports) {
+        for (const port of props.ports) {
+          const protocol = port.protocol || "tcp";
+          portMappings[`${port.external}`] = `${port.internal}/${protocol}`;
+        }
+      }
+
+      // Prepare volume mappings
+      const volumeMappings: Record<string, string> = {};
+      if (props.volumes) {
+        for (const volume of props.volumes) {
+          const readOnlyFlag = volume.readOnly ? ":ro" : "";
+          volumeMappings[volume.hostPath] =
+            `${volume.containerPath}${readOnlyFlag}`;
+        }
+      }
+
+      // Create new container
+      const containerId = await api.createContainer(imageRef, containerName, {
+        ports: portMappings,
+        env: props.environment,
+        volumes: volumeMappings,
+        cmd: props.command,
+      });
+
+      // Connect to networks if specified
+      if (props.networks) {
+        for (const network of props.networks) {
+          const networkId =
+            typeof network === "string" ? network : network.name;
+          await api.connectNetwork(containerId, networkId, {
+            aliases: network.aliases,
+          });
+        }
```

Comment:
do you need to diff `props.networks` with `this.props.networks` (the old props) and disconnect from networks on `update`?

Reply:
I will try changing the props and see what happens.